src/main/java/de/tum/in/www1/artemis/domain/ILearningObject.java

```diff
@@ -2,8 +2,9 @@
 
 import java.time.ZonedDateTime;
 import java.util.Optional;
+import java.util.Set;
 
-public interface Completable {
+public interface ILearningObject {
```

Comment:
Seams like you only allow certain inheritance from this interface further down. You might want to think about marking this interface as sealed, which is a new feature of Java 17

Reply:
Thanks for pointing this out! I'd have to move the classes to the same package for this (_"Class is not allowed to extend sealed class from another package"_), but I like the current structure.